Tidy up window initial position code

XXX: Can we have overrideRedirect windows which are transientFor?
If so, do we want them to be parented or not?  The side effect of
this change is not to parent them...

XXX: So where does this end up positioning windows which aren't
CW_USEDEFAULT?

---
 xserver/hw/xwin/winmultiwindowwindow.c |   24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

Index: xorg-server-1.5.3/xserver/hw/xwin/winmultiwindowwindow.c
===================================================================
--- xorg-server-1.5.3.orig/xserver/hw/xwin/winmultiwindowwindow.c	2009-02-16 12:29:40.843750000 +0000
+++ xorg-server-1.5.3/xserver/hw/xwin/winmultiwindowwindow.c	2009-02-16 12:30:43.937500000 +0000
@@ -508,17 +508,6 @@
   iX = pWin->drawable.x + GetSystemMetrics (SM_XVIRTUALSCREEN);
   iY = pWin->drawable.y + GetSystemMetrics (SM_YVIRTUALSCREEN);
 
-  /* Default positions if none specified */
-  if (!winMultiWindowGetWMNormalHints(pWin, &hints))
-    hints.flags = 0;
-  if ( !(hints.flags & (USPosition|PPosition)) &&
-       !winMultiWindowGetTransientFor (pWin, NULL) &&
-       !pWin->overrideRedirect )
-    {
-      iX = CW_USEDEFAULT;
-      iY = CW_USEDEFAULT;
-    }
-
   iWidth = pWin->drawable.width;
   iHeight = pWin->drawable.height;
 
@@ -574,6 +563,8 @@
   wc.lpszClassName = pszClass;
   RegisterClassEx (&wc);
 
+  if (!pWin->overrideRedirect)
+  {
     if (winMultiWindowGetTransientFor (pWin, &pDaddy))
     {
       if (pDaddy)
@@ -582,6 +573,17 @@
         if (hFore && (pDaddy != (WindowPtr)GetProp(hFore, WIN_WID_PROP))) hFore = NULL;
       }
     }
+    else
+    {
+      /* Default positions if none specified */
+      if (!winMultiWindowGetWMNormalHints(pWin, &hints)) hints.flags = 0;
+      if (!(hints.flags & (USPosition|PPosition)))
+      {
+        iX = CW_USEDEFAULT;
+        iY = CW_USEDEFAULT;
+      }
+    }
+  }
 
   /* Create the window */
   /* Make it OVERLAPPED in create call since WS_POPUP doesn't support */
